Calling win.webContents.printToPDF multiple times

Calling win.webContents.printToPDF causes a crash the second time I do it.

The code is simply:
1)I have some html, i call ipc.Main.on, this has a callback function, inside I have a win.webContents.printToPDF and a callback function for it. If I call it the first time everything goes smooth, then if I change dinamically my HTML page with javascript and try to call the ipc.Main.on again the win.webContents.printToPDF gives me an error and crashes my application:

npm ERR! code ELIFECYCLE
npm ERR! errno 3221225477
npm ERR! lavoro@1.0.0 start: `electron .`
npm ERR! Exit status 3221225477
npm ERR!
npm ERR! Failed at the lavoro@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\(MY NAME)\AppData\Roaming\npm-cache\_logs\2019-09-28T23_17_42_726Z-debug.log

--- a/lib/browser/api/web-contents.js
+++ b/lib/browser/api/web-contents.js
@@ -71,7 +71,6 @@ const defaultPrintingSetting = {
   headerFooterEnabled: false,
   marginsType: 0,
   isFirstRequest: false,
-  requestID: getNextId(),
   previewUIID: 0,
   previewModifiable: true,
   printToPDF: true,
@@ -204,7 +203,10 @@ WebContents.prototype.executeJavaScript = function (code, hasUserGesture) {
 
 // Translate the options of printToPDF.
 WebContents.prototype.printToPDF = function (options) {
-  const printingSetting = Object.assign({}, defaultPrintingSetting)
+  const printingSetting = {
+    ...defaultPrintingSetting,
+    requestID: getNextId()
+  }
   if (options.landscape) {
     printingSetting.landscape = options.landscape
   }

--- a/spec-main/api-web-contents-spec.ts
+++ b/spec-main/api-web-contents-spec.ts
@@ -1400,6 +1400,19 @@ describe('webContents module', () => {
       const data = await w.webContents.printToPDF({})
       expect(data).to.be.an.instanceof(Buffer).that.is.not.empty()
     })
+
+    it('does not crash when called multiple times', async () => {
+      const w = new BrowserWindow({ show: false, webPreferences: { sandbox: true } })
+      await w.loadURL('data:text/html,<h1>Hello, World!</h1>')
+      const promises = []
+      for (let i = 0; i < 2; i++) {
+        promises.push(w.webContents.printToPDF({}))
+      }
+      const results = await Promise.all(promises)
+      for (const data of results) {
+        expect(data).to.be.an.instanceof(Buffer).that.is.not.empty()
+      }
+    })
   })
 
   describe('PictureInPicture video', () => {

